src/routes/credentials.rs

```diff
@@ -2,6 +2,8 @@ use std::collections::HashMap;
 
 use actix_session::Session;
 use actix_web::{get, post, web, HttpResponse};
+use base64::Engine;
```

Comment:
Can we rename this whole file `authentication`? We can't change the endpoint name without introducing breaking changes, but the name `credentials` is not appropriate, in my opinion.